content/shared/extended-plugin-api.md

```diff
@@ -0,0 +1,319 @@
+        if value < 0:
+            raise ValueError(f"uint64 field '{key}' cannot be negative")
+        self.fields[key] = f"{value}u"
+        return self
+
+    def int64_field(self, key: str, value: int) -> 'LineBuilder':
+        """Add an integer field to the line protocol."""
+        self._validate_key(key, "field")
+        self.fields[key] = f"{value}i"
+        return self
+
+    def float64_field(self, key: str, value: float) -> 'LineBuilder':
+        """Add a float field to the line protocol."""
+        self._validate_key(key, "field")
+        # Check if value has no decimal component
+        self.fields[key] = f"{int(value)}.0" if value % 1 == 0 else str(value)
+        return self
+
+    def string_field(self, key: str, value: str) -> 'LineBuilder':
+        """Add a string field to the line protocol."""
+        self._validate_key(key, "field")
+        # Escape quotes and backslashes in string values
+        escaped_value = value.replace('"', '\\"').replace('\\', '\\\\')
+        self.fields[key] = f'"{escaped_value}"'
+        return self
+
+    def bool_field(self, key: str, value: bool) -> 'LineBuilder':
+        """Add a boolean field to the line protocol."""
+        self._validate_key(key, "field")
+        self.fields[key] = 't' if value else 'f'
+        return self
+
+    def time_ns(self, timestamp_ns: int) -> 'LineBuilder':
+        """Set the timestamp in nanoseconds."""
+        self._timestamp_ns = timestamp_ns
+        return self
+
+    def build(self) -> str:
+        """Build the line protocol string."""
+        # Start with measurement name (escape commas only)
+        line = self.measurement.replace(',', '\\,')
+
+        # Add tags if present
+        if self.tags:
+            tags_str = ','.join(
+                f"{k}={v}" for k, v in self.tags.items()
+            )
+            line += f",{tags_str}"
+
+        # Add fields (required)
+        if not self.fields:
+            raise InvalidLineError(f"At least one field is required: {line}")
+
+        fields_str = ','.join(
+            f"{k}={v}" for k, v in self.fields.items()
+        )
+        line += f" {fields_str}"
+
+        # Add timestamp if present
+        if self._timestamp_ns is not None:
+            line += f" {self._timestamp_ns}"
+
+        return line
+```
+{{% /expand %}}
+{{% /expand-wrapper %}}
+
+#### Query data
+
+Your plugins can execute SQL queries and process results directly:
+
+```python
+# Simple query
+results = influxdb3_local.query("SELECT * FROM metrics WHERE time > now() - INTERVAL '1 hour'")
+
+# Parameterized query for safer execution
+params = {"table": "metrics", "threshold": 90}
+results = influxdb3_local.query("SELECT * FROM $table WHERE value > $threshold", params)
+```
+
+Query results come back as a `List` of `Dict[String, Any]`, where each dictionary represents a row with column names as keys and column values as values. This makes it easy to process the results in your plugin code.
```

Comment:
```suggestion
Query results are a `List` of `Dict[String, Any]`, where each dictionary represents a row with column names as keys and column values as values.
```

We usually want to avoid making claims (for example, "makes it easy"). This one's not bad, but I would replace it with a brief example or just state that you can use standard Python to process the results.